cmd/mirror-main.go

```diff
@@ -748,7 +748,7 @@ func runMirror(ctx context.Context, cancelMirror context.CancelFunc, srcURL, dst
 	// Check if we are only trying to mirror one bucket from source.
 	if dstClt.GetURL().Type == objectStorage &&
 		dstClt.GetURL().Path == string(dstClt.GetURL().Separator) && !mirrorAllBuckets {
-		dstURL = urlJoinPath(dstURL, srcClt.GetURL().Path)
+		dstURL = urlJoinPath(dstURL, filepath.Base(srcClt.GetURL().Path))
```

Comment:
We should copy top-level directories from the source directory into the destination object storage as buckets. But this change mirrors everything to a single bucket.

Reply:
@krisis,
Fixed.